vm.c

```diff
@@ -2382,7 +2428,9 @@ vm_exec_loop(rb_execution_context_t *ec, enum ruby_tag_type state,
     rb_ec_raised_reset(ec, RAISED_STACKOVERFLOW | RAISED_NOMEMORY);
     while (UNDEF_P(result = vm_exec_handle_exception(ec, state, result))) {
         /* caught a jump, exec the handler */
-        result = vm_exec_core(ec);
+        if (UNDEF_P(result = jit_exec_exception(ec))) {
+            result = vm_exec_core(ec);
+        }
```

Comment:
Could you add a comment to explain what `result == undef` means in this context?

Reply:
sure, added a comment in https://github.com/ruby/ruby/pull/8171/commits/0bc8f420a22157a06ae2bb340441ffd3ccd0b173